apps/core/app/[locale]/(default)/cart/_components/item-quantity.tsx

```diff
@@ -126,38 +122,87 @@ const parseSelectedOptions = (selectedOptions: CartItemData['selectedOptions'])
+const SubmitButton = ({ children, ...props }: ComponentPropsWithoutRef<'button'>) => {
+  const { pending } = useFormStatus();
+  const t = useTranslations('Cart.SubmitItemQuantity');
 
-  const [counterValue, setCounterValue] = useState<'' | number>(quantity);
-  const handleCountUpdate = async (value: string | number) => {
-    if (value === '') {
-      setCounterValue(value);
+  return (
+    <button
+      className="hover:text-primary focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-primary/20 disabled:text-gray-200"
+      disabled={pending}
+      type="submit"
+      {...props}
+    >
+      {children}
+      {pending && <span className="sr-only">{t('spinnerText')}</span>}
+    </button>
+  );
+};
 
-      return;
-    }
+const Quantity = ({ value }: { value: number }) => {
+  const { pending } = useFormStatus();
+  const t = useTranslations('Cart.SubmitItemQuantity');
 
-    setCounterValue(Number(value));
+  return (
+    <span className="flex w-10 justify-center">
+      {pending ? (
+        <>
+          <Spinner aria-hidden="true" className="animate-spin text-primary" />
+          <span className="sr-only">{t('spinnerText')}</span>
+        </>
+      ) : (
+        <span>{value}</span>
+      )}
+    </span>
+  );
+};
 
-    const productData: UpdateProductQuantityData = Object.assign(
-      {
-        lineItemEntityId: entityId,
-        productEntityId,
-        quantity: Number(value),
-        selectedOptions: parseSelectedOptions(selectedOptions),
-      },
-      variantEntityId && { variantEntityId },
-    );
+export const ItemQuantity = ({ product }: { product: Product }) => {
+  const t = useTranslations('Cart.SubmitItemQuantity');
 
-    await updateProductQuantity(productData);
+  const { quantity, entityId, productEntityId, variantEntityId, selectedOptions } = product;
+  const [productQuantity, setProductQuantity] = useState<number>(quantity);
+
+  useEffect(() => {
+    setProductQuantity(quantity);
+  }, [quantity]);
+
+  const onSubmit = async (formData: FormData) => {
+    const { status } = await updateItemQuantity({
+      lineItemEntityId: entityId,
+      productEntityId,
+      quantity: Number(formData.get('quantity')),
+      selectedOptions: parseSelectedOptions(selectedOptions),
+      variantEntityId,
+    });
+
+    if (status === 'error') {
+      toast.error(t('errorMessage'), {
+        icon: <AlertCircle className="text-error-secondary" />,
+      });
+    }
   };
 
   return (
-    <Counter
-      className="w-32 text-base font-bold"
-      min={1}
-      onChange={handleCountUpdate}
-      value={counterValue}
-    />
+    <div className="border-2 border-gray-200 p-2.5">
+      <form action={onSubmit} className="flex items-center">
+        <SubmitButton
+          aria-label={t('submitReduceText')}
+          onClick={() => setProductQuantity(productQuantity - 1)}
+        >
+          <Minus aria-hidden="true" />
+        </SubmitButton>
```

Comment:
🍹 Use `aria-*` attributes as a last resort.
```suggestion
        <SubmitButton onClick={() => setProductQuantity(productQuantity - 1)}>
          <Minus>
            <title>{t('submitReduceText')}</title>
          </Minus>
        </SubmitButton>
```